### ISSUE TITLE
Define mechanism for game authors to specify inventory conditions

### ISSUE THREAD

**Issue by Dumplingz** [uchicago-cs/chiventure#786]
Issues #236, #577, and #772 implement a new kind of condition that checks a player's inventory for a specific item.

Some background info:
An inventory condition has only two inputs: a `player_t` with an inventory to check, and an `item_t` to match in the inventory, which both exist in the condition files in game-state. However, #577 and #772 radically change the way a condition is saved in-memory, so it is important to note that now there is a general struct `condition_t` that saves both an attribute condition and an inventory condition. A condition is only tied to an action/end condition through a linkedlist of conditions, which are added to the functions `add_end_condition_to_game` and `add_condition` in game.c (located in game-state).

Currently, WDL has no way of saving this kind of inventory condition.

Issue #236 will be complete after this is implemented.

**Comment by sabinecodes15** [comment]
The branch wdl/inventory-condition was created to implement the features in this issue. In this [commit](https://github.com/uchicago-cs/chiventure/commit/f930bc87a86191c731bc41e66464b577fc8d1cad), I updated `load_actions()` in `load_items.c` to check for action conditions. Still need to make sure the condition is being implemented. In addition, I created a [sample wdl file](https://github.com/uchicago-cs/chiventure/commit/f930bc87a86191c731bc41e66464b577fc8d1cad#diff-6ff09b9896dfe1bca5df291f383da06e05a4bd47dc9f8a203a3e6b1e71e5753b) that implements a condition for the TAKE action for testing purposes.

**Comment by sabinecodes15** [comment]
We realized that the updated implementation of load_actions() was failing because the function build_conditions() operates under the assumption that an item has already been added to the game; thus, I had been trying to add conditions to actions too early. I instead [added a load_conditions() function](https://github.com/uchicago-cs/chiventure/commit/c8056122bae5b415a40173cc21e0b49f4bd6853b) that iterates through all the actions of an item _after_ it has been added to the game, then adds conditions to actions that contain a "conditions" attribute. This seems to be working in the test wdl file I made (not having the chair prevented me from taking the table, and having the chair permitted me to do so). The next step is to write tests to ensure the conditions are being loaded properly.

**Comment by sabinecodes15** [comment]
Upon closer inspection, we realized that implementing tests for inventory conditions should be a separate issue. Thus, we will close this issue with the approval of PR #1492.

**Comment by sabinecodes15** [comment]
Closing statement:

The implementation of `load_conditions()` provides a way for game developers to optionally specify conditions for item actions in a WDL by first verifying that a "conditions" object exists for an action, then iterating through the object to append each condition to the action's conditions list. We also generalized the function in a way that should permit the smooth integration of attribute and level conditions in the future. Now that this has been implemented, we can close this issue.
